browser-test/src/session_timeout.test.ts

```diff
@@ -0,0 +1,167 @@
+import {test, expect} from './support/civiform_fixtures'
+import {
+  enableFeatureFlag,
+  loginAsAdmin,
+  loginAsTestUser,
+  validateScreenshot,
+  validateToastMessage,
+} from './support'
+import {waitForHtmxReady} from './support/wait'
+import {
+  setupSessionTimeout,
+  verifyTimeoutModalBehavior,
+  verifyModalDismissal,
+  verifyAutoLogout,
+  advanceToSessionInactivityWarning,
+  advanceToSessionLengthWarning,
+  advanceToSessionTimeout,
+} from './support/session_timeout'
+
+test.describe('Session timeout warnings for Admin', () => {
+  test.beforeEach(async ({page, adminSettings}) => {
+    await setupSessionTimeout({page, adminSettings})
+    await loginAsAdmin(page)
+    await page.goto('/admin/programs')
+    await page.waitForLoadState('networkidle')
+  })
+
+  test('shows inactivity warning modal, extends session, show session length warning modal and logout', async ({
+    page,
+  }) => {
+    await test.step('Show and verify inactivity warning', async () => {
+      await advanceToSessionInactivityWarning(page)
+      await page.waitForSelector(
+        '#session-inactivity-warning-modal div.usa-modal',
+        {state: 'visible', timeout: 20000},
+      )
```

Comment:
```suggestion
      await expect(page.locator('#session-inactivity-warning-modal div.usa-modal')).toBeVisible()
```

(blocking) Use the web first assertions instead of waitForSelector. Like `networkidle` there should almost never be a reason to manually set timeouts. Playwright will handle waiting and will timeout based on global defaults.

Anything with explicit timeout needs to be updated.